src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -2213,6 +2251,83 @@ METHOD(kernel_ipsec_t, destroy, void,
 	free(this);
 }
 
+struct tunnel_counter_key {
+	uint32_t terminator_id;
+	const char* tenant_id;
+	const char* account_id;
+	const char* vrf_name;
+};
+
+static bool tunnel_counter_key_compare(const void *key, const void *other_key) {
+    struct tunnel_counter_key *tunnel_1 = (struct tunnel_counter_key *)key;
+    struct tunnel_counter_key *tunnel_2 = (struct tunnel_counter_key *)other_key;
+
+	return tunnel_1->terminator_id == tunnel_2->terminator_id &&
+			!strcmp(tunnel_1->tenant_id, tunnel_2->tenant_id) && 
+			!strcmp(tunnel_1->account_id, tunnel_2->account_id) && 
+			!strcmp(tunnel_1->vrf_name, tunnel_2->vrf_name);
+}
+
+static u_int tunnel_counter_key_hash(const void *key) {
+	struct tunnel_counter_key *tunnel_key = (struct tunnel_counter_key *)key;
+	char tunnel_key_str[MAX_ACCOUNT_ID_LEN + MAX_TENANT_ID_LEN + MAX_VRF_NAME_LEN + 10];
+	sprintf(tunnel_key_str, "%u%s%s%s", tunnel_key->terminator_id, tunnel_key->tenant_id, tunnel_key->account_id, tunnel_key->vrf_name);
+	return hashtable_hash_str(&tunnel_key_str);
```

Comment:
No need to reinvent the wheel regarding combining hashes of multiple elements, there's already support for it:
```
return chunk_hash_inc(chunk_from_thing(tunnel_key->terminator_id),
           chunk_hash_inc(chunk_from_str((char*)tunnel_key->tenant_id),
               chunk_hash_inc(chunk_from_str((char*)tunnel_key->account_id),
                   chunk_hash(chunk_from_str((char*)tunnel_key->vrf_name)))));
```

This will result in better hash distribution than the current one (which can have unnecessary collisions due to the concatenation, e.g. ` terminator_id = 1, tenant_id = "a", account_id = "b", vrf_name = "cd"` and `terminator_id = 1, tenant_id = "a", account_id = "bc", vrf_name = "d"` will result in the same hash here) and there is much less room for errors (for example you forgot to add +1 to the buffer size to account for the NULL terminator)